Nuxt tutorial (#1325)
denoland/docs: examples/tutorials/nuxt.md

---
title: "Build a Nuxt app with Deno"
url: /examples/nuxt_tutorial/
---

[Nuxt](https://nuxt.com/) is a framework that provides an intuitive way to
create full-stack applications based on [Vue](https://vuejs.org/). It offers

> file-based routing, server-side rendering capabilities, and automatic code

Reviewer: One of the benefits of Nuxt is how you can mix lots of different rendering models. So I suggest this small edit here:

```suggestion
file-based routing, a variety of rendering options, and automatic code
```

splitting out of the box. With its modular architecture, Nuxt simplifies the
development process by providing a structured approach to building Vue
applications.

In this tutorial, we'll build a simple Nuxt application with Deno that will
display a list of dinosaurs and allow you to learn more about each one when you
click on the name:

- [Scaffold a Nuxt app](#scaffold-a-nuxt-app-with-deno)
- [Setup server API routes](#setup-server-api-routes)
- [Setup Vue frontend](#setup-vue-frontend)
- [Add Tailwind](#add-tailwind)
- [Next steps](#next-steps)

You can find the code for this project in this
[repo](https://github.com/denoland/examples/tree/main/with-nuxt).

## Scaffold a Nuxt app with Deno

We can create a new Nuxt project using deno like this:

```bash
deno -A npm:nuxi@latest init
```

> Here, we’re using npm as the package manager. This will create a nuxt-app with

Reviewer: Is that the best language when talking about the package manager? Might it be more correct to say that we're using Deno as the package manager, and using the package from NPM?

These might be silly semantics, but I'm thinking of landing the point over time, that deno is a package manager too.

So, perhaps...

```suggestion
We'll use Deno to manage our package dependencies, and can grab the Nuxt package from npm. This will create a nuxt-app with
```

```tsx
// layouts/default.vue

<template>
  <div>
    <slot />
  </div>
</template>;
```

In this project, we’re also going to use [tailwind](https://tailwindcss.com/)
for some basic design, so we need to install those dependencies:

```bash
deno install -D npm:tailwindcss npm:postcss npm:autoprefixer
```

Then initialize Tailwind:

```bash
deno -A npm:tailwindcss init
```

This will create a `tailwind.config.js` file in the root of our directory. Let’s
open that up and update the `content` section to include key files in our
project:

```tsx
// tailwind.config.js

/** @type {import('tailwindcss').Config} */
export default {
  content: [
    "./components/**/*.{js,vue,ts}",
    "./layouts/**/*.vue",
    "./pages/**/*.vue",
    "./plugins/**/*.{js,ts}",
    "./app.vue",
  ],
  theme: {
    extend: {},
  },
  plugins: [],
};
```

Next, let’s add the tailwind utilities to a new css file, `assets/css/main.css`:

```tsx
// assets/css/main.css

base;
components;
utilities;
```

The only other thing we'll need to do is update our `nuxt.config.ts` file to
configure our Nuxt application for Deno compatibility, enable development tools,

> and set up Tailwind CSS. We're disabling SSR for better Deno compatibility and

Reviewer: Is there a way to phrase this that implies less of a limitation for Deno? It sounds a but like Deno can't do SSR. It might be a touch ambiguous if we mean the runtime or Deno Deploy when we say that too.

Maybe...

```suggestion
and set up Tailwind CSS. We're disabling SSR and instead generating the entire site for the greatest flexibility in deployment options, and
```
